src/spyglass/spikesorting/analysis/v1/unit_annotation.py

```diff
@@ -0,0 +1,150 @@
+            """Fetch the spike times for a restricted set of units
+
+            Parameters
+            ----------
+            return_unit_ids : bool, optional
+               whether to return unit ids with spike times, by default False
+
+            Returns
+            -------
+            list of np.ndarray
+                list of spike times for each unit in the group,
+                if return_unit_ids is False
+            tuple of list of np.ndarray, list of str
+                list of spike times for each unit in the group and the unit ids,
+                if return_unit_ids is True
+            """
+            return (UnitAnnotation & self).fetch_unit_spikes(return_unit_ids)
+
+    def add_annotation(self, key, **kwargs):
+        """Add an annotation to a unit. Creates the unit if it does not exist.
+
+        Parameters
+        ----------
+        key : dict
+            dictionary with key for Annotation
+
+        Raises
+        ------
+        ValueError
+            if unit_id is not valid for the sorting
+        """
+        # validate new units
+        unit_key = {
+            "spikesorting_merge_id": key["spikesorting_merge_id"],
+            "unit_id": key["unit_id"],
+        }
+        if not self & unit_key:
+            nwb_file = (
+                SpikeSortingOutput & {"merge_id": key["spikesorting_merge_id"]}
+            ).fetch_nwb()[0]
+            nwb_field_name = self._get_spike_obj_name(nwb_file)
+            spikes = nwb_file[nwb_field_name]["spike_times"].to_list()
+            if key["unit_id"] > len(spikes):
+                raise ValueError(
+                    f"unit_id {key['unit_id']} is greater than ",
+                    f"the number of units in {key['spikesorting_merge_id']}",
+                )
+            self.insert1(unit_key)
+        # add annotation
+        self.Annotation().insert1(key, **kwargs)
+
+    def fetch_unit_spikes(self, return_unit_ids=False):
+        """Fetch the spike times for a restricted set of units
+
+        Parameters
+        ----------
+        return_unit_ids : bool, optional
+           whether to return unit ids with spike times, by default False
+
+        Returns
+        -------
+        list of np.ndarray
+            list of spike times for each unit in the group,
+            if return_unit_ids is False
+        tuple of list of np.ndarray, list of str
+            list of spike times for each unit in the group and the unit ids,
+            if return_unit_ids is True
+        """
+        if len(self) == len(UnitAnnotation()):
+            logger.warning(
+                "fetching all unit spikes",
+                "if this is unintended, please call as: ",
+                "(UnitAnnotation & key).fetch_unit_spikes()",
+            )
+        # get the set of nwb files to load
+        merge_keys = [
+            {"merge_id": merge_id}
+            for merge_id in list(set(self.fetch("spikesorting_merge_id")))
```

Comment:
I think this table could be assigned the value of `merge_alias` to avoid continually rewriting `spikesorting_merge_id`. This convention could be adopted across similar tables to make it easier to deal with the aliasing enforced by strict adherence to `merge_id` pks for merge tables

Reply:
To check my understanding: are you suggesting to define a variable `merge_alias = "spikesorting_merge_id"` at the top of the file and using that, or something more datajoint-esque I'm missing?